vortex-array/src/array/bool/compute/cast.rs

```diff
@@ -0,0 +1,20 @@
+use vortex_dtype::DType;
+use vortex_error::VortexResult;
+
+use crate::array::{BoolArray, BoolEncoding};
+use crate::compute::CastFn;
+use crate::{Array, IntoArray};
+
+impl CastFn<BoolArray> for BoolEncoding {
+    fn cast(&self, array: &BoolArray, dtype: &DType) -> VortexResult<Array> {
+        assert!(matches!(dtype, DType::Bool(_)));
```

Comment:
You can't just panic here, return a result explaining why you can't perform this cast.